src/tests/test_resultset.py

```diff
@@ -635,3 +637,48 @@ def test_doesnt_refresh_sqlaproxy_if_different_connection():
     list(first_set)
 
     assert id(first_set._sqlaproxy) == original_id
+
+
+@pytest.mark.parametrize(
+    "function, expected_warning",
+    [
+        (
+            "pie",
+            (
+                ".pie() is deprecated and will be removed in a future version. "
+                "Use %sqlplot pie instead. "
+                "For more help, find us at https://ploomber.io/community "
+            ),
+        ),
+        (
+            "bar",
+            (
+                ".bar() is deprecated and will be removed in a future version. "
+                "Use %sqlplot bar instead. "
+                "For more help, find us at https://ploomber.io/community "
+            ),
+        ),
+        (
+            "plot",
+            (
+                ".plot() is deprecated and will be removed in a future version. "
+                "For more help, find us at https://ploomber.io/community "
+            ),
+        ),
+    ],
+)
+def test_deprecated_warnings(config, function, expected_warning):
+    with warnings.catch_warnings(record=True) as record:
```

Comment:
inside the `with` statement, you should only include the line that triggers the warning, otherwise it's hard for readers to understand where the warning should come from

Reply:
Noted, will restructure

Reply:
[Fixed] https://github.com/ploomber/jupysql/pull/870/files#diff-d64bdb5f9c7dbd521843544e6ff605632e011c98ab5e2c3078e1bcc66de9337cR680-R683